README.md

```diff
@@ -15,6 +15,193 @@ Documentation
+        (.tertiaryLabel, .secondarySystemBackground, .normalText),
+        // tertiary label on tertiary background
+        (.tertiaryLabel, .tertiarySystemBackground, .normalText),
+
+        // system red on system background (fails)
+        // (.systemRed, .systemBackground, .normalText),
+    ]
+
+    let allColorSpaces: [UITraitCollection] = [
+        // Light Mode
+        UITraitCollection(userInterfaceStyle: .light),
+        // Light Mode, Increased Contrast
+        UITraitCollection(traitsFrom: [
+            UITraitCollection(userInterfaceStyle: .light),
+            UITraitCollection(accessibilityContrast: .high)
+        ]),
+        // Dark Mode
+        UITraitCollection(userInterfaceStyle: .dark),
+        // Dark Mode, Increased Contrast
+        UITraitCollection(traitsFrom: [
+            UITraitCollection(userInterfaceStyle: .dark),
+            UITraitCollection(accessibilityContrast: .high)
+        ])
+    ]
+
+    func testColorContrast() {
+        // test across all color modes we support
+        for traits in allColorSpaces {
+            // test each color pair
+            colorPairs.forEach {
+                let color1 = $0.foreground.resolvedColor(with: traits)
+                let color2 = $0.background.resolvedColor(with: traits)
+
+                XCTAssertTrue(
+                    color1.isSufficientContrast(to: color2, context: $0.context, level: .AA),
+                    String(
+                        format: "#%@ vs #%@ ratio = %.02f under %@ Mode%@",
+                        color1.rgbDisplayString(),
+                        color2.rgbDisplayString(),
+                        color1.contrastRatio(to: color2),
+                        traits.userInterfaceStyle == .dark ? "Dark" : "Light",
+                        traits.accessibilityContrast == .high ? " Increased Contrast" : ""
+                    )
+                )
+            }
+        }
+    }
+}
+```
+
+### 3. UIScrollView extensions to assist with keyboard avoidance
+
+#### FormViewController
+
+ `FormViewController` is a view controller with a scrollable content area that will automatically avoid the keyboard for you. It is a good choice for views that have inputs (e.g. login or onboarding). Even for views without inputs, it is still quite useful for managing the creation of a `UIScrollView` and a `contentView` set within it, so that you can focus on your content and not have to code a scrollView for every view. 
+
+#### UIScrollview Extensions
+
+Want to have a scrollview that avoids the keyboard, but you can’t use `FormViewController`? Most of its functionality is a simple extension to `UIScrollView`. You can add keyboard avoidance to any scroll view like so:
+
+```swift
+scrollView.registerKeyboardNotifications()
+```
+<aside>
+💡 Almost every full-screen view in your app that contains any text should be a scroll view because of the vagaries of localization, Dynamic Type, potentially small screen sizes, and landscape mode support.
```

Comment:
```suggestion
💡 Almost every full-screen view in your app that contains any text should be a vertical scroll view because of the vagaries of localization, Dynamic Type, potentially small screen sizes, and landscape mode support.
```
A clarification that we should add here.